### PR TITLE
Fix User Role in People Section

### LINKED ISSUE #2455: Bug Report: User Role Not Displaying in 'People' Section
**Describe the bug**
In the 'People' section of the Talawa User Portal, the "Role" column does not display any roles for the users in some cases (as shown in the first screenshot). This is inconsistent with other organizations where roles such as "Superadmin" or "User" are properly displayed (refer to the second screenshot).

**To Reproduce**

1. Log in to the Talawa User Portal.
2. Navigate to the 'People' section in an organization (e.g., Unity Foundation).
3. Observe the "Role" column — roles are missing for all users.

**Expected behavior**
The "Role/User-Type" column should display the correct role (e.g., Admin, Superadmin, User) for each user in the table.

**Actual behavior**
The "Role/User-Type" column is empty, leaving users and admins without role visibility

**Screenshots**
Screenshot 1: (Role missing)
image
Screenshot 2: (Role displayed correctly)
image

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by aditya-bansal-7**
**What kind of change does this PR introduce?**
Bug fix

**Issue Number:**

Fixes #2455

**Did you add tests for your changes?**

No

**Snapshots/Videos:**

https://github.com/user-attachments/assets/eaf40f64-b7a8-4300-af43-153b0f8de2a7

**If relevant, did you update the documentation?**

no

**Summary**
This PR addresses the issue in the Talawa User Portal where the "Role" column is not displaying the correct roles for users in the 'People' section for some organizations

**Does this PR introduce a breaking change?**
no

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes

## Summary by CodeRabbit

- **New Features**
	- Enhanced member management with improved filtering options for 'All Members' and 'Admins'.
	- Introduced a new state variable to better handle member roles and types.
	- Added constants for filter modes to streamline functionality.

- **Bug Fixes**
	- Improved error handling for organization loading failures.

- **Tests**
	- Added a new test to verify that members are rendered with the correct user type.

### CODE REVIEW

**Comment by palisadoes** [comment]
1. Please make the changes requested so that CodeRabbit.ai approves the PR
2. You have not written any tests for the code. The code coverage test will fail. You need to make sure that you get to 100% code coverage for the files you changed. This is what the patch test means.

File: src/screens/UserPortal/People/People.test.tsx [hunk 1]
```diff
@@ -222,4 +222,22 @@ describe('Testing People Screen [User Portal]', () => {
...
+            </I18nextProvider>
+          </Provider>
+        </BrowserRouter>
+      </MockedProvider>,
+    );
+
+    await wait();
+    expect(screen.getByText('Admin')).toBeInTheDocument();
+    expect(screen.getByText('User')).toBeInTheDocument();
+  });
```
[the remark this answers is not in the document]
Comment (aditya-bansal-7) [reply src/screens/UserPortal/People/People.test.tsx:242]: ? This will not work roles are not in table

**Review by aditya-bansal-7 — COMMENTED** [review]
review changes

**Comment by aditya-bansal-7** [comment]
@palisadoes review this

**Comment by palisadoes** [comment]
> @palisadoes review this

Please read:
- https://github.com/PalisadoesFoundation/talawa-admin/pull/2459#issuecomment-2489413567

**Comment by aditya-bansal-7** [comment]
@palisadoes check

**Comment by palisadoes** [comment]
> @palisadoes check

PTAL
- https://github.com/PalisadoesFoundation/talawa-admin/pull/2459#issuecomment-2489413567

**Comment by Cioppolo14** [comment]
There isn’t an issue assigned to you for this PR. Please follow the guidelines in our PR_GUIDELINES.md file. We have the procedures in place so that everyone has a fair chance of contributing. I will be closing this pull request. Please follow the procedures and resubmit when ready.

**Pull request closed without merging**